### PR TITLE
Allow users to cancel individual tickets within an order

### LINKED ISSUE #3072: Redesign ticket cancellation flow to support cancellation of individual tickets within an order
The current ticket cancellation flow only allows users to cancel an entire order. This is too restrictive in cases where an order contains multiple tickets and the user only wants to cancel one or some of them.

A redesign of the cancellation system is required so that cancellations can be handled at ticket level, not only at order level.

Image

**Current behavior**

* Users can cancel an entire order
* Users cannot cancel an individual ticket within an order
* Multi-ticket orders therefore require full cancellation even when only one ticket should be canceled

**Expected behavior**

* Allow cancellation of individual tickets within an order
* Keep the remaining tickets in the same order active and valid
* Reflect ticket-level cancellation clearly in the UI and backend state
* Ensure order totals, payment status, invoice data, and availability are updated correctly after partial cancellation

**Why this is needed**

* Supports real-world ticketing scenarios where users buy multiple tickets in one order
* Prevents unnecessary full-order cancellations
* Improves flexibility and user experience
* Reduces manual support work for organizers

**Required redesign areas**

* **Data model**

  * Support cancellation status at ticket or attendee level, not only order level
  * Preserve overall order integrity when one or more tickets are canceled

* **User flow**

  * Add UI for selecting which ticket(s) to cancel
  * Show clear distinction between:

    * cancel full order
    * cancel selected ticket(s)

* **Organizer flow**

  * Allow organizers to view and manage partial cancellations
  * Ensure exports and attendee management reflect partially canceled orders correctly

* **Financial logic**

  * Recalculate totals when only part of an order is canceled
  * Define how refunds are handled for partial cancellations
  * Ensure payment and invoice records remain consistent

* **Inventory logic**

  * Released ticket quantity should return to availability where applicable
  * Capacity and quota tracking must update correctly for partial cancellations

* **Communication**

  * Update confirmation and cancellation emails for partial cancellation cases
  * Clearly communicate which ticket(s) were canceled and which remain valid

**Edge cases to handle**

* Orders with multiple ticket types
* Orders with mixed paid and free tickets
* Orders with attendee-specific data already submitted
* Partial cancellation after payment completion
* Partial cancellation with discount codes, taxes, fees, or donations
* Orders containing tickets that are no longer cancellable due to policy or timing rules

**Scope**

* Redesign cancellation architecture for ticket-level handling
* Add frontend and backend support for partial cancellation
* Update order management UI, attendee UI, and organizer views
* Adjust exports, emails, and reporting to reflect partial cancellations

**Out of scope**

* No automatic refund policy changes unless explicitly defined
* No change to cancellation policies themselves unless needed as part of implementation

**Additional notes**

* Backward compatibility for existing orders should be considered carefully
* It may be useful to define separate states such as:

  * active
  * partially canceled
  * fully canceled
* This likely requires coordinated updates across orders, attendees, payments, invoices, and ticket availability logic

### PR DESCRIPTION
**Pull request by Aqil-Ahmad**
fixes #3072 

## What was added

- New model properties on `Order`: `user_cancelable_positions`, `user_partial_cancel_allowed`, 
  and `user_partial_cancel_fee` for position-level cancellation logic
- A new Celery task `cancel_order_positions` that handles cancellation, fee, and refund logic 
  at the individual ticket level
- Two new presale views (`OrderPositionCancel`, `OrderPositionCancelDo`) and URL routes 
  for the partial cancel flow
- UI changes on the order detail page - a "Cancel individual tickets" button and a banner 
  showing how many tickets have already been canceled
- A new template `order_cancel_positions.html` for selecting tickets to cancel with 
  estimated fees and refund amounts

## Edge cases handled

- Base tickets with non-cancelable addons are excluded from the cancelable list
- Cancellation fees are calculated proportionally based on the canceled amount
- Addon prices are correctly included when calculating canceled totals
- Inherits existing order eligibility checks (expiry, checkin status, approval mode, etc.)
- All cancellations are attributed to the user in the audit log

## Tests added

- Model test for addon-based cancelability exclusion
- Presale view tests: unknown order 404s, pending order cancellation, paid order 
  cancellation with fee verification

https://github.com/user-attachments/assets/6e41b5c4-d311-4fa2-82c6-d15700daaf54

https://github.com/user-attachments/assets/6191881c-17c0-4d0c-afeb-4d662bad5248

https://github.com/user-attachments/assets/364cbe63-773f-43a7-8e78-a1c0cff9104b

## Summary by Sourcery

Add support for user-initiated cancellation of individual tickets within an order, including UI, backend logic, and proportional cancellation fee calculation.

New Features:
- Allow customers to cancel one or more selected tickets in an order while keeping remaining tickets valid.
- Expose a dedicated order page for selecting cancelable tickets with per-ticket refund and fee estimates.
- Support optional issuing of refunds for canceled tickets as gift cards or back to the original payment method on partial cancellation.

Enhancements:
- Refine order cancellation eligibility logic to distinguish between full-order and partial ticket cancellations and to account for non-cancelable add-ons.
- Introduce proportional calculation of cancellation fees for partial order cancellations, considering existing cancellation fees.
- Surface information on already-canceled tickets within an order on the order detail page.

Tests:
- Add tests covering access control and 404 handling for the new partial cancellation endpoints.
- Add tests verifying correct behavior of partial cancellation for pending and paid orders, including refund amounts and cancellation fees.
- Add model-level tests ensuring user_cancelable_positions excludes base tickets with non-cancelable add-ons.

### CODE REVIEW

File: app/eventyay/base/models/orders.py [hunk 1]
```diff
@@ -569,6 +569,96 @@ def user_cancel_fee(self):
...
+            .select_related('product')
+            .prefetch_related('issued_gift_cards')
+        )
+
+        cancelable = []
+        for op in positions:
+            if not op.product.allow_cancel or op.has_checkin:
+                continue
+            if any(gc.value != op.price for gc in op.issued_gift_cards.all()):
+                continue
```
[the remark this answers is not in the document]
Comment (Aqil-Ahmad) [reply app/eventyay/base/models/orders.py]: fixed.

File: app/eventyay/presale/templates/pretixpresale/event/order_cancel_positions.html [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+{% load i18n %}
+{% load eventurl %}
+{% load money %}
+{% load presale_locale %}
+{% block title %}{% trans 'Cancel selected tickets' %}{% endblock %}
+{% block content %}
+    <h2>
+        {% blocktrans trimmed with code=order.code %}
+            Cancel tickets in order: {{ code }}
+        {% endblocktrans %}
```
[the remark this answers is not in the document]
Comment (Aqil-Ahmad) [reply app/eventyay/presale/templates/pretixpresale/event/order_cancel_positions.html:11]: it'll  bring inconsistencies in formatting.

File: app/tests/tickets/presale/test_orders.py [hunk 3]
```diff
@@ -479,6 +493,98 @@ def test_orders_cancel_paid(self):
...
+    def test_orders_cancel_positions_paid_with_refund_and_fee(self):
+        with scopes_disabled():
+            keep_pos = OrderPosition.objects.create(
+                order=self.order,
+                item=self.ticket,
+                variation=None,
+                price=Decimal('23.00'),
+                attendee_name_parts={'full_name': 'Alice'},
+            )
+            self.order.status = Order.STATUS_PAID
```
[the remark this answers is not in the document]
Comment (Aqil-Ahmad) [reply app/tests/tickets/presale/test_orders.py]: a single-position free order wouldn't reach the partial cancel flow. Same for the full cancel, canceling all positions is blocked, users must use the existing full-order cancel flow for that case.

The current tests cover the core partial cancellation path. These edge cases are intentionally unsupported flows.

**Comment by Aqil-Ahmad** [comment]
@Sak1012 could you please review?

File: app/eventyay/presale/views/order.py [hunk 4]
```diff
@@ -980,6 +985,99 @@ def dispatch(self, request, *args, **kwargs):
...
+            return redirect(self.get_order_url())
+        return super().dispatch(request, *args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        ctx = super().get_context_data(**kwargs)
+        cancelable_ids = [p.pk for p in self.order.user_cancelable_positions]
+        positions = (
+            self.order.positions.filter(pk__in=cancelable_ids)
+            .select_related('product', 'variation', 'addon_to')
+            .prefetch_related('addons')
```
[the remark this answers is not in the document]
Comment (Aqil-Ahmad) [reply app/eventyay/presale/views/order.py]: done

File: app/eventyay/presale/views/order.py [hunk 5]
```diff
@@ -980,6 +986,95 @@ def dispatch(self, request, *args, **kwargs):
...
+    def get_context_data(self, **kwargs):
+        ctx = super().get_context_data(**kwargs)
+        positions = self.order.user_cancelable_positions
+        prefetch_related_objects(positions, 'variation', 'addon_to', 'addons')
+        positions = sorted(positions, key=lambda p: p.positionid)
+
+        position_rows = []
+        for position in positions:
+            addons = list(position.addons.all())
+            position_total = position.price + sum(addon.price for addon in addons)
```
[the remark this answers is not in the document]
Comment (Aqil-Ahmad) [reply app/eventyay/presale/views/order.py]: position_total = base.price + addon_prices is the correct representation.

File: app/eventyay/base/models/orders.py [hunk 6]
```diff
@@ -569,6 +569,80 @@ def user_cancel_fee(self):
...
+
+        if self.status != Order.STATUS_PAID or self.total <= Decimal('0.00'):
+            return Decimal('0.00')
+
+        fee = self.user_cancel_fee
+        if fee <= Decimal('0.00'):
+            return Decimal('0.00')
+
+        proportional_fee = round_decimal(fee * canceled_total / self.total, self.event.currency)
+        return min(canceled_total, proportional_fee)
```
[the remark this answers is not in the document]
Comment (Aqil-Ahmad) [reply app/eventyay/base/models/orders.py]: necessary tests were added after the feature was implemented

**Comment by Rachit7168** [comment]
@codex review

**Comment by Rachit7168** [comment]
Please Resolve outstanding reviews from codex

File: app/eventyay/base/services/orders.py [hunk 7]
```diff
@@ -2690,6 +2691,140 @@ def _try_auto_refund(
...
+
+            selected_positions[position_id] = cancelable_positions[position_id]
+
+        positions_to_cancel = []
+        canceled_total = Decimal('0.00')
+        canceled_ids = set()
+        # Sorting by position ID keeps parent positions before their add-ons.
+        for position in sorted(selected_positions.values(), key=lambda p: p.positionid):
+            if position.addon_to_id and position.addon_to_id in selected_positions:
+                continue
```
[the remark this answers is not in the document]
Comment (Aqil-Ahmad) [reply app/eventyay/base/services/orders.py:2822]: fixed

File: app/eventyay/base/models/orders.py [hunk 8]
```diff
@@ -569,6 +569,114 @@ def user_cancel_fee(self):
...
+
+        fee = self.user_cancel_fee - self.user_existing_cancellation_fee
+        if fee <= Decimal('0.00'):
+            return Decimal('0.00')
+
+        remaining_total = self.total - self.user_existing_cancellation_fee
+        if remaining_total <= Decimal('0.00'):
+            return Decimal('0.00')
+
+        proportional_fee = round_decimal(fee * canceled_total / remaining_total, self.event.currency)
```
[the remark this answers is not in the document]
Comment (Aqil-Ahmad) [reply app/eventyay/base/models/orders.py]: the current formula is consistent with how user_cancel_fee works for full order cancellation, but i'll look further into this and maybe i will create a follow-up pr if it needed

**Comment by Aqil-Ahmad** [comment]
with stripe

https://github.com/user-attachments/assets/89781871-184c-4f9a-ad90-5a6e3f3a18e1

**Comment by Aqil-Ahmad** [comment]
@Sak1012 could you please take a look and approve the overall flow for paid tickets, i'll then resolve some of the comments and do other improvements

File: app/eventyay/presale/views/order.py [hunk 9]
```diff
@@ -413,6 +415,9 @@ def get_context_data(self, **kwargs):
         ).exclude(provider__in=('offsetting', 'reseller', 'boxoffice', 'manual'))
         ctx['user_change_allowed'] = self.order.user_change_allowed
         ctx['user_cancel_allowed'] = self.order.user_cancel_allowed
+        ctx['user_partial_cancel_allowed'] = self.order.user_partial_cancel_allowed
+        canceled_positions_qs = self.order.all_positions.filter(canceled=True)
```
[the remark this answers is not in the document]
Comment (Aqil-Ahmad) [reply app/eventyay/presale/views/order.py]: fixed.

**Review by Rachit7168 — APPROVED** [review]
Have tested, looks good to me!

File: app/eventyay/presale/views/order.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (Aqil-Ahmad) [reply app/eventyay/presale/views/order.py]: fixed

File: app/eventyay/base/services/orders.py [hunk 10]
```diff
@@ -2690,6 +2691,146 @@ def _try_auto_refund(
...
+
+            if position.pk not in canceled_ids:
+                canceled_total += position.price
+                canceled_ids.add(position.pk)
+
+            for addon in addons:
+                if addon.pk in canceled_ids:
+                    continue
+                canceled_total += addon.price
+                canceled_ids.add(addon.pk)
```
[the remark this answers is not in the document]
Comment (Aqil-Ahmad) [reply app/eventyay/base/services/orders.py]: fixed

**Comment by ViRUS-0-0** [comment]
Hi @Aqil-Ahmad , I just checked out your PR in my local dev. I seem to get the following error.

https://github.com/user-attachments/assets/d3f0c8a7-e4d2-4a4c-94c0-c58cf97abfd7

Could you please confirm it?

**Comment by Aqil-Ahmad** [comment]
@ViRUS-0-0 could you please test it again. It works for me

image

**Review by mariobehling — CHANGES_REQUESTED** [review]
The changes show the flow for the user, but we also need to show "cancelled" in the organiser views. This should not remove a ticket, instead a ticket should show as cancelled and it should be possible to reinstate it by an organiser if this was done in error. Please implement this as discussed in meetings previously.

**Comment by Aqil-Ahmad** [comment]
@mariobehling canceled tickets already show in the organizer order detail view, they appear in red (not removed) with the cancellation logged in the order history. Individual position reinstatement by the organizer needs to be implemented i'll get it done shortly.

image

**Comment by Aqil-Ahmad** [comment]
@mariobehling please have a look!

https://github.com/user-attachments/assets/1b063da8-5db0-407f-851c-fc66d9c90c3a

**Comment by Aqil-Ahmad** [comment]
@Sak1012 @mariobehling fixed all the issues we discussed

image

**Review by ArnavBallinCode — CHANGES_REQUESTED** [review]
please change the string as discussed in the meeting

**Comment by Aqil-Ahmad** [comment]
image

image

File: app/eventyay/static/pretixcontrol/scss/main.scss [hunk 11]
```diff
@@ -744,7 +744,13 @@ details summary {
...
 .pos-canceled * {
     color: $brand-danger;
-    text-decoration: line-through !important;
+    text-decoration: line-through;
+}
+
+.pos-canceled .position-buttons,
+.pos-canceled .position-buttons * {
+    text-decoration: none;
+    color: $text-color;
```
[the remark this answers is not in the document]
Comment (Aqil-Ahmad) [reply app/eventyay/static/pretixcontrol/scss/main.scss:772]: not worth fixing

File: app/eventyay/control/templates/pretixcontrol/order/index.html [hunk 12]
```diff
@@ -380,6 +380,18 @@ <h3 class="panel-title">
                                         {% endif %}
                                         {% eventsignal event "eventyay.control.signals.order_position_buttons" order=order position=line request=request %}
                                     </div>
+                                {% else %}
+                                    {% if order.status == 'p' or order.status == 'n' %}
+                                        <div class="position-buttons">
+                                            <form method="post"
+                                                  action="{% url "control:event.order.position.reinstate" event=request.event.slug organizer=request.event.organizer.slug code=order.code position=line.pk %}">
```
[the remark this answers is not in the document]
Comment (Aqil-Ahmad) [reply app/eventyay/control/templates/pretixcontrol/order/index.html:387]: leaving for consistency

File: app/eventyay/presale/templates/pretixpresale/event/order_cancel_positions.html [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+        </p>
+
+        <div class="panel panel-default">
+            <div class="panel-heading">
+                <strong>{% trans 'Tickets in this order' %}</strong>
+            </div>
+            <ul class="list-group">
+                {% for row in position_rows %}
+                    <li class="list-group-item">
+                        <div class="checkbox" style="margin: 0;">
```
Comment (prayag78) [inline app/eventyay/presale/templates/pretixpresale/event/order_cancel_positions.html]: please avoid using inline styles

**Review by prayag78 — CHANGES_REQUESTED** [review]
Tested, works!

File: app/eventyay/presale/templates/pretixpresale/event/order_cancel_positions.html (hunk 13 above)
Comment (Aqil-Ahmad) [reply app/eventyay/presale/templates/pretixpresale/event/order_cancel_positions.html]: Ohhh!! Missed that one. Let me fix

**Comment by Aqil-Ahmad** [comment]
@prayag78 thankss !!! for the review, addressed your concern. any further issues you noticed?

**Review by prayag78 — APPROVED** [review]

**Review by sarafarajnasardi — APPROVED** [review]

**Comment by hpdang** [comment]
@Saksham-Sirohi why is this "On Hold"?

**Comment by ArnavBallinCode** [comment]
> "On Hold"?

It was on hold at that time because it was directly impacting the normal ticketing flow, I think it is good to go now. @Saksham-Sirohi

**Review by Saksham-Sirohi — COMMENTED** [review]
## Review: Allow users to cancel individual tickets within an order

Important customer-facing feature. The position-level cancelability logic is thoughtfully implemented.

### Strengths
- `user_cancelable_positions` handles complex rules: check-ins, gift cards, addon bundles (base ticket not cancelable if any addon isn't).
- `user_partial_cancel_fee` prorates flat/percentage cancellation fees against already-charged fees — avoids double-charging on multiple partial cancels.
- Respects `cancel_allow_user_paid_require_approval` by disabling partial cancel when approval mode is on (only full-order requests supported) — correct product constraint.
- `is_partially_canceled` property for UI state.

### Concerns

1. **Race conditions** — between page load and cancel submit, a position could be checked in. Confirm the cancel endpoint re-validates `user_cancelable_positions` inside a transaction, not just trusts client-selected position IDs.

2. **Quota release** — partial cancel should return quota for canceled positions. Verify waiting list isn't incorrectly triggered (compare with full cancel path's `count_waitinglist` behavior).

3. **Payment/refund flow** — for paid orders, document whether partial cancel issues automatic refund or creates a refund request. UI should show fee breakdown before confirm.

4. **API parity** — if presale is web-only, fine; if mobile/API clients exist, check whether partial cancel needs an API endpoint.

5. **Test coverage** — given order model complexity, tests should cover: addon bundle blocked, gift card partial value blocked, fee proration across two partial cancels, check-in after page load rejected.

**Verdict:** Strong domain logic; needs transactional re-validation and quota/refund tests before merge.

**Review by Saksham-Sirohi — COMMENTED** [review]
## Review: Partial ticket cancellation within an order

Solid domain modeling for a complex feature.

### Strengths
- `user_cancelable_positions` correctly handles check-ins, gift cards, and addon bundles (base ticket blocked if any addon isn't cancelable).
- `user_partial_cancel_fee` prorates flat/percentage fees against already-charged cancellation fees — avoids double-charging on repeated partial cancels.
- Disables partial cancel when `cancel_allow_user_paid_require_approval` is on — matches the approval-only full-order flow.

### Suggestions
1. **Re-validate at submit** — re-run cancelability checks inside the cancel endpoint transaction; a position could be checked in between page load and POST.
2. **Quota release** — confirm canceled positions return quota with the same `count_waitinglist` semantics as full-order cancel.
3. **Refund UX** — for paid orders, surface fee breakdown before confirm; document whether refund is immediate or approval-based.
4. **Tests** — addon bundle blocked, gift-card partial value blocked, fee proration across two partial cancels, check-in race rejected.

Comment only — no approval from my side.

**Review by Saksham-Sirohi — COMMENTED** [review]
## Review: Partial ticket cancellation within an order

Thorough implementation — the cancel service re-validates selected positions against a fresh `user_cancelable_positions` map at submit time, so check-in races between page load and POST are already handled.

### Strengths
- Addon bundle rule: base ticket not cancelable if any addon isn't.
- `user_partial_cancel_fee` prorates against prior cancellation fees.
- Partial cancel correctly disabled when paid-order approval mode is on.
- Tests added for addon exclusion (`test_user_cancelable_positions_excludes_base_with_noncancelable_addon`).

### Remaining suggestions
1. **Quota release** — worth confirming canceled positions return quota with the same semantics as full-order cancel.
2. **Paid-order UX** — fee breakdown before confirm (template already computes `cancellation_fee`; verify it's shown clearly).

Comment only.

**Comment by Aqil-Ahmad** [comment]
> ### Concerns
> 1. **Race conditions** — between page load and cancel submit, a position could be checked in. Confirm the cancel endpoint re-validates `user_cancelable_positions` inside a transaction, not just trusts client-selected position IDs.

confirmed, its already handled

> 2. **Quota release** — partial cancel should return quota for canceled positions. Verify waiting list isn't incorrectly triggered (compare with full cancel path's `count_waitinglist` behavior).

verified. and its correctly handled

> 3. **Payment/refund flow** — for paid orders, document whether partial cancel issues automatic refund or creates a refund request. UI should show fee breakdown before confirm.

Fee breakdown: already shown in the template before confirm
Auto-refund: cancel_order_positions calls _try_auto_refund when try_auto_refund=True, which attempts immediate refund if the payment provider supports it, otherwise creates a manual refund request. This mirrors full cancel.

> 4. **API parity** — if presale is web-only, fine; if mobile/API clients exist, check whether partial cancel needs an API endpoint.

can be a follow-up feature, but we dont have api clients/ mobile app using it

> 5. **Test coverage** — given order model complexity, tests should cover: addon bundle blocked, gift card partial value blocked, fee proration across two partial cancels, check-in after page load rejected.

> **Verdict:** Strong domain logic; needs transactional re-validation and quota/refund tests before merge.

File: app/eventyay/presale/templates/pretixpresale/event/order_cancel_positions.html [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+                <strong>{% trans 'Tickets in this order' %}</strong>
+            </div>
+            <ul class="list-group">
+                {% for row in position_rows %}
+                    <li class="list-group-item">
+                        <div class="checkbox">
+                            <label>
+                                <input type="checkbox" name="positions" value="{{ row.position.pk }}">
+                                <strong>#{{ row.position.positionid }}</strong>
+                                {{ row.position.product.name|event_localize }}
```
Comment (ViRUS-0-0) [inline app/eventyay/presale/templates/pretixpresale/event/order_cancel_positions.html:34]: The page displays tickets and their add-ons in a flat list of checkboxes. Selecting a base ticket automatically cancels all of its active add-ons under the hood, but there is no visual warning or grouped nesting indicating this behavior. A user might select a base ticket, leave its add-on unchecked, and be surprised when both are canceled

Consider adding a brief warning notice in the template to make it clear that canceling a base ticket will also cancel its associated add-ons

**Review by ViRUS-0-0 — CHANGES_REQUESTED** [review]

**Comment by Aqil-Ahmad** [comment]
@ViRUS-0-0 done.

image

**Review by ArnavBallinCode — CHANGES_REQUESTED** [review]
This PR now allows customers to cancel add-ons individually, we've exposed a pre-existing bug in `OrderChangeManager`'s `CancelOperation` that we should probably fix here. 

Currently, when a parent ticket is canceled, the operation loops over **all** of its add-ons to cancel them via `for opa in op.position.addons.all():`. Because of the `.all()` call, this includes add-ons that have already been canceled. 

If a customer cancels an add-on first and later cancels the parent ticket, the system will process the already-canceled add-on a second time (e.g., decrementing voucher usages again and duplicating log entries). 


2. There is a critical copy-paste bug in the `CancelOperation` logic for add-ons that becomes much more prominent now that we are heavily supporting partial and add-on cancellations. 

When an add-on is canceled as part of a parent ticket's cancellation, the system attempts to invalidate its ticket secret. However, it passes the parent ticket (`op.position`) to `assign_ticket_secret` instead of the add-on (`opa`).

**Review by ArnavBallinCode — COMMENTED** [review]
Basic functionality is not working as tested in meeting, please make a fresh pr with fresh demo when completely ready as discussed in the meeting.

**Comment by Aqil-Ahmad** [comment]
> Basic functionality is not working as tested in meeting, please make a fresh pr with fresh demo when completely ready as discussed in the meeting.

which basic functionality isn't working? I think we discussed another PRs needs to be created with the current changes just squashed so its clean.

**Comment by Rachit7168** [comment]
Cancel order with addons was not working discussed in the meet. Please address that as well !
